### PR TITLE
regex selection of multiple branches

### PR DESCRIPTION
**Pull request by aminnj**
Similarly to `uproot`, allows 
```julia
f = ROOTFile("test/samples/NanoAODv5_sample.root")
LazyTree(f, "Events", [r"Muon_(pt|eta|phi)$", "Muon_charge", "Muon_pt"])
```
to select branches
```julia 
[:Muon_pt, :Muon_eta, :Muon_phi, :Muon_charge]
```
(Non-regex requires an exact match)

### CODE REVIEW

**Comment by Moelf** [comment]
can we have some synthetic test on the `_m` 3 lines? (I guess just recreate those 3 lines in test if you don't want to pull them outside.

**Comment by Moelf** [comment]
also I guess we can add
```
LazyTree(f::ROOTFile, s::AbstractString, branch::Union{AbstractString, Regex) =
LazyTree(f, s, [branch])
```
? In case people want to construct from a single Regex (or String...)

File: test/runtests.jl [hunk 1]
```diff
@@ -252,6 +252,10 @@ end
     HLT_Mu3_PFJet40 = UnROOT.array(rootfile, "Events/HLT_Mu3_PFJet40")
     @test eltype(HLT_Mu3_PFJet40) == Bool
     @test HLT_Mu3_PFJet40[1:3] == [false, true, false]
+    tree = LazyTree(rootfile, "Events", [r"Muon_(pt|eta|phi)$", "Muon_charge", "Muon_pt"])
+    @test Set(propertynames(tree)) == Set([:Muon_pt, :Muon_eta, :Muon_phi, :Muon_charge])
```
Comment (Moelf) [inline test/runtests.jl:256]:
```suggestion
    @test propertynames(tree) == [:Muon_pt, :Muon_eta, :Muon_phi, :Muon_charge]
```

File: test/runtests.jl [hunk 2]
```diff
@@ -252,6 +252,10 @@ end
...
+    tree = LazyTree(rootfile, "Events", r"Muon_(pt|eta)$")
+    @test Set(propertynames(tree)) == Set([:Muon_pt, :Muon_eta])
```
Comment (Moelf) [inline test/runtests.jl:258]:
```suggestion
    @test propertynames(tree) == [:Muon_pt, :Muon_eta]
```

**Review by Moelf — CHANGES_REQUESTED** [review]

File: test/runtests.jl (hunk 1 above)
Comment (Moelf) [reply test/runtests.jl:256]: in the unlikely event we have duplication.... `Set` will prevent it from erroring I guess

**Comment by Moelf** [comment]
thanks a lot! we want all the quality of life improvement before 1.0!

**Comment by tamasgal** [comment]
Nice, thanks! :)

**Pull request merged**